app/uk/gov/hmrc/cataloguefrontend/healthindicators/HealthIndicatorsController.scala

```diff
@@ -41,20 +41,14 @@ class HealthIndicatorsController @Inject()(
       }
     }
 
-  def indicatorsForAllRepos(): Action[AnyContent] =
+  def indicatorsForRepoType(repoType: String): Action[AnyContent] =
     Action.async { implicit request =>
-      for {
-        repoRatingsWithTeams  <- healthIndicatorsService.createRepoRatingsWithTeams
-        form      = HealthIndicatorsFilter.form.bindFromRequest
-        repoTypes = repoRatingsWithTeams.map(_.repositoryType).distinct
-        filteredIndicators = form.fold(_ => None, _.repoType) match {
-          case Some(rt) => repoRatingsWithTeams.filter(_.repositoryType == rt)
-          case None     => repoRatingsWithTeams
-        }
-      } yield
-        filteredIndicators match {
-          case repoRatingsWithTeams => Ok(HealthIndicatorsLeaderBoard(repoRatingsWithTeams, form, repoTypes))
-        }
+      RepoType.parse(repoType).fold(_ => Future.successful(Redirect(routes.HealthIndicatorsController.indicatorsForRepoType(RepoType.Service.asString))),
+        r => for {
+          repoRatingsWithTeams <- healthIndicatorsService.findRepoRatingsWithTeams(Some(r))
+          form = HealthIndicatorsFilter.form.bindFromRequest
```

Comment:
Is this form needed now we get repoType as a query param?

Reply:
I think so. When removed the search stopped working correctly